clusterman/draining/queue.py

```diff
@@ -248,7 +251,20 @@ def process_drain_queue(
                     self.submit_host_for_termination(host_to_process)
             elif host_to_process.scheduler == "kubernetes":
                 logger.info(f"Kubernetes host to drain and submit for termination: {host_to_process}")
-                self.submit_host_for_termination(host_to_process, delay=0)
+                try:
+                    if kube_operator_client:
+                        k8s_drain(
+                            kube_operator_client, host_to_process.hostname,
+                        )
+                    else:
+                        logger.info(
+                            f"Unable to drain {host_to_process.hostname} (no Kubernetes connector configured) - "
+                            "continuing to terminate anyway!",
+                        )
+                except Exception as e:
+                    logger.error(f"Failed to drain {host_to_process.hostname} continuing to terminate anyway: {e}")
```

Comment:
We would like create something to churning/recycling instances to keep them fresh. 
CEP1869 - https://yelpwiki.yelpcorp.com/display/SEC/Project%3A+Kubernetes+Instance+Upgrade
We planned to protect applications with PDB. Draining should respects protection logic.
You used  pod eviction what is perfect choice for this. But we would like to clean-up pods fully before termination. Because application can be corrupted/worked with bad performance if we don't respect PDB. 
I think we should discuss this in #kubernetes-instance-upgrade, please ping me once you have availability :)